File: backend/Core/Infrastructure/Refund.cs
namespace Core.Infrastructure
{
    public class Refund
    {
        public Guid Id { get; set; }
        public Guid TenantId { get; set; }
        public Guid TransactionId { get; set; }
        public Guid? CustomerId { get; set; }
        public string StripeRefundId { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public string Currency { get; set; } = "usd";
        public string? Reason { get; set; }
        public string? Notes { get; set; }
        public string Status { get; set; } = "pending";
        public string? ApprovedBy { get; set; }
        public DateTime? ApprovedAt { get; set; }
        public DateTime? ProcessedAt { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public Tenant Tenant { get; set; }
        public PaymentTransaction Transaction { get; set; }
        public Customer Customer { get; set; }
    }
}

File: frontend/src/types/refund.ts
export interface Refund {
  id: string;
  transactionId: string;
  customerId: string | null;
  customerName: string;
  customerEmail: string;
  stripeRefundId: string;
  amount: number;
  currency: string;
  reason: string;
  notes: string;
  status: string;
  approvedBy: string;
  approvedAt: string | null;
  processedAt: string | null;
  createdAt: string;
}

export interface RefundStats {
  totalRefunds: number;
  totalAmount: number;
  refundRate: number;
  pendingCount: number;
  avgProcessingTimeHours: number;
}
